pkg/utils/executors.go

```diff
@@ -219,10 +202,7 @@ var ValidMacOSAppleSiliconResourceClasses = []string{
 	"macos.m1.large.gen1",
 }
 
-var ValidMacOSResourceClasses = slices.Concat(
-	ValidMacOSAppleSiliconResourceClasses,
-	ValidMacOSIntelResourceClasses,
-)
+var ValidMacOSResourceClasses = ValidMacOSAppleSiliconResourceClasses
```

Comment:
Since Intel macs are no longer supported, only Apple Silicon, we could probably simplify this even further by removing the mention to Apple Silicon, and not creating a useless variable alias. What do you think of just having:

```go
var ValidXcodeVersions = []string{
	"15.4.0",
	"15.3.0",
	"15.2.0",
	"15.1.0",
	"15.0.0",
	"14.3.1",
	"14.2.0",
	"14.1.0",
	"14.0.1",
	"13.4.1",
}

var ValidMacOSResourceClasses = []string{
	"macos.m1.medium.gen1",
	"macos.m1.large.gen1",
}
```

Additionally, I see [in the docs](https://circleci.com/docs/using-macos/#supported-xcode-versions-silicon) that Xcode `16.0.0` is supported. This PR could probably add that to `ValidXcodeVersions`.